### PR TITLE
Heroku app json

### PR DESCRIPTION
**Pull request by jotes**
I accidentally force pushed my master branch.
This is a continuation of https://github.com/mozilla/pontoon/pull/446 .
- [x] provide an dummy auth provider because we can't support firefox accounts safely

Ref. bug:
https://bugzilla.mozilla.org/show_bug.cgi?id=1316171

### CODE REVIEW

**Comment by jotes** [comment]
@mathjazz Hi, could you review this integration again?

File: docs/admin/deployment.rst [hunk 1]
```diff
@@ -12,12 +12,13 @@ Buildpack
 ---------
 Pontoon uses several buildpacks in a specific order. They are (in order):
 
-1. `heroku-buildpack-apt`_ for installing Subversion.
-2. `heroku-buildpack-ssh`_ for setting up the SSH keys necessary for committing
+1. `heroku-buildpack-submodules` to fetch all related git submodules.
```
Comment (mathjazz) [inline docs/admin/deployment.rst]: Just curious, how do submodules work right now, without this buildpack?

File: pontoon/settings/base.py [hunk 2]
```diff
@@ -50,7 +54,29 @@ def path(*args):
...
+            app_url = "https://{}".format(Site.objects.get(pk=1).domain)
+            cache.set(APP_URL_KEY, app_url)
+
+        return app_url
+
+    SITE_URL = lazy(_site_url, str)()
+else:
+    SITE_URL = os.environ.get('SITE_URL', 'http://localhost:8000')
+
+BROWSERID_AUDIENCES = [SITE_URL]
```
Comment (mathjazz) [inline pontoon/settings/base.py]: We no longer need `BROWSERID_AUDIENCES`. It was used by Persona and was removed from master.

File: pontoon/base/views.py [hunk 3]
```diff
@@ -1343,3 +1346,31 @@ def user_settings(request):
         'available_locales': available_locales,
         'selected_locales': selected_locales,
     })
+
+
+def heroku_setup(request):
+    """
+    Heroku don't allow us to set SITE_URL or Site during the build phase of an app.
```
Comment (mathjazz) [inline pontoon/base/views.py]: doesn't :-)

File: app.json [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+       "plan": "heroku-postgresql",
+       "options": {
+         "version": "9.5"
+       }
+     }
+   ],
+   "env": {
+     "ADMIN_EMAIL": {
+       "value": "pontoon@example.com",
+       "description": "An email of the administrator."
```
Comment (mathjazz) [inline app.json]: We should just reuse descriptions from `docs/admin/deployment.rst`.

File: docs/admin/deployment.rst [hunk 5]
```diff
@@ -12,12 +12,13 @@ Buildpack
 ---------
```
Comment (mathjazz) [inline docs/admin/deployment.rst]: We should add the `Deploy to Heroku` button under the first paragraph of this file.

Start with a sentence like this: *For quick and easy deployment without leaving your web browser, click this buttton:*. And then add a button.

File: docs/admin/deployment.rst (hunk 1 above)
Comment (jotes) [inline docs/admin/deployment.rst]: @mathjazz I think that David Ascher had the same problems: https://bugzilla.mozilla.org/show_bug.cgi?id=1286062

File: pontoon/settings/base.py (hunk 2 above)
Comment (jotes) [inline pontoon/settings/base.py]: You're right, that's a leftover from the previous rebase.

File: app.json (hunk 4 above)
Comment (jotes) [inline app.json]: 'k

**Pull request merged**